core/templates/dev/head/domain/statistics/PlaythroughImprovementTaskObjectFactory.ts

```diff
@@ -50,16 +51,22 @@ angular.module('oppia').factory('PlaythroughImprovementTaskObjectFactory', [
               'Mark as Resolved', 'btn-danger')
               .result.then(() => PlaythroughIssuesService.resolveIssue(issue))
               .then(() => this._isObsolete = true);
-          }),
+          },
+          () => this._isEnabled),
```

Comment:
Why do we pass this in as a function and not a boolean? 

Also, shouldn't the availability be known at the time of creation of this list? I.e. we should determine availability  first and then populate the list, right? We shouldn't show the button at all (even a disabled one) if the user has no edit rights.

Reply:
> Why do we pass this in as a function and not a boolean?

Was a design decision -- for some buttons the status of "is enabled" may change through interaction with the task.

> Also, shouldn't the availability be known at the time of creation of this list? I.e. we should determine availability first and then populate the list, right?

Yes, I just wasn't able to understand how to access the UserService's fields without relying on async logic.

> We shouldn't show the button at all (even a disabled one) if the user has no edit rights.

I don't think that's worthwhile at the moment since the card will look "bad", and I don't think it's too big of a deal to show a disabled button anyway.

Reply:
> Yes, I just wasn't able to understand how to access the UserService's fields without relying on async logic.

Ah -- I meant that you would populate the buttons array only after the async callback from UserService has been received. Would that work?

> I don't think that's worthwhile at the moment since the card will look "bad", and I don't think it's too big of a deal to show a disabled button anyway.

I don't understand -- why would it look bad? I believe that we already do this in the rest of the exploration editor (i.e. hide the buttons entirely and make things non-clickable), so we should follow that practice (of not exposing any buttons) here as well.

Reply:
Done